### PR TITLE
Workflow updates for GLDAS fixes

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR brings in the workflow updates to address GLDAS enhancements requested by NCO. GLDAS script updates come in via a new GLDAS repo tag [gldas_gfsv16_release.v.2.1.1](https://github.com/NOAA-EMC/GLDAS/releases/tag/gldas_gfsv16_release.v.2.1.1).

Workflow side updates:

- New GLDAS tag `gldas_gfsv16_release.v.2.1.1`
- Update ecFlow definition file to remove the `jgdas_atmos_gldas` job from the 06z, 12z, and 18z cycles; includes removing it from the forecast job triggers for those cycles as well
- Created release notes
- Updated GFS version to v16.3.17 (tentative, will adjust after receiving confirmation from NCO)

Refs #2503

# Type of change

Production update

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

Visual inspection, not yet tested on WCOSS2.

### CODE REVIEW

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA The other jobs can still move on even if the 00cycle gldas fails? Or we need to set the gldas job complete manually to move other jobs. If the latter is true, probably we should skip the gldas if CPC gauge data is missing.

**Review by aerorahul — APPROVED** [review]
looks good.
thanks.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA The other jobs can still move on even if the 00cycle gldas fails? Or we need to set the gldas job complete manually to move other jobs. If the latter is true, probably we should skip the gldas if CPC gauge data is missing.

@HelinWei-NOAA I defer to you on this but will give you my thoughts: If the job isn't critical to the GFS running, then I would not exit with a non-zero number. From NCO's perspective I imagine they'd prefer the job to not fail if data is missing, print a warning that the data is missing and job will not run, and exit with 0. The alternative is to have the job fail but if the missing data is not recoverable then the job shouldn't fail and hold up production. It should essentially be skipped.

> probably we should skip the gldas if CPC gauge data is missing

I tend to agree. The job should still run but if the data is missing it should print the warning and exit gracefully.

Based on what we're talking about, [these lines](https://github.com/NOAA-EMC/GLDAS/blob/release/gfs.v16.3_gldas/ush/gldas_forcing.sh#L55:L56) of `ush/gldas_forcing.sh` should be changed to something like this:
```
cpc=$pathp1/$cpc_precip
if [ ! -s $cpc ]; then cpc=$pathp2/$cpc_precip ; fi
if [ $RUN_ENVIR = "nco" ]; then cpc=$pathp2/$cpc_precip ; fi
if [ ! -s $cpc ]; then 
 echo "WARNING: GLDAS MISSING $cpc, JOB WILL NOT RUN."
 exit 0
fi
```

@aerorahul your thoughts?

**Comment by HelinWei-NOAA** [comment]
The sole factor influencing GLDAS to generate different initial land states than gdas is CPC gauge data. Running GLDAS without this data is unnecessary. Therefore, I propose implementing this restriction at a higher level, such as in gfs_v16_3.def. Without the data, the task jgdas_atmos_gldas should not be initiated.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA Understood, however, ecFlow does not have data dependencies (like rocoto does). We can adjust triggers based on the status of other jobs but can't (to my knowledge) fire off and/or set a job to complete based on data availability. We need to handle the data availability within the job. 99.9% of the time the data is available so I think it would be fine to always let the job run but just exit with a warning and `exit 0` if the data is missing.

File: ecf/defs/gfs_v16_3.def
Comment (HelinWei-NOAA) [inline ecf/defs/gfs_v16_3.def:1]: IF CPC GAUGE DATA IS MISSING AND THE GLDAS JOB EXIT WITH 2, ./atmos/init/jgdas_atmos_gldas == complete IN LINE 2374 IS STILL SET TO BE TRUE?

**Comment by HelinWei-NOAA** [comment]
Sounds good. I have another question. If CPC gauge data is missing and the gldas job exit with 2, ./atmos/init/jgdas_atmos_gldas  in line 2374 of gfs_v16_3.def is still set to complete?
> @HelinWei-NOAA Understood, however, ecFlow does not have data dependencies (like rocoto does). We can adjust triggers based on the status of other jobs but can't (to my knowledge) fire off and/or set a job to complete based on data availability. We need to handle the data availability within the job. 99.9% of the time the data is available so I think it would be fine to always let the job run but just exit with a warning and `exit 0` if the data is missing.

**Comment by KateFriedman-NOAA** [comment]
> I have another question. If CPC gauge data is missing and the gldas job exit with 2, ./atmos/init/jgdas_atmos_gldas in line 2374 of gfs_v16_3.def is still set to complete?

Only `exit 0` will lead to a "complete" job. Any exit # other than 0 will fire off an `ecflow_client --abort` signal and message that there was an error running the job. See this section of the GLDAS ecf script:

```
${HOMEgfs}/jobs/JGDAS_ATMOS_GLDAS
if [ $? -ne 0 ]; then
   ecflow_client --msg="***JOB ${ECF_NAME} ERROR RUNNING J-SCRIPT ***"
   ecflow_client --abort
   exit
fi
```
https://github.com/NOAA-EMC/global-workflow/blob/dev/gfs.v16/ecf/scripts/gdas/atmos/init/jgdas_atmos_gldas.ecf#L48:L53

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA The gldas task (/ecf/scripts/gdas/atmos/init/jgdas_atmos_gldas.ecf) uses ${HOMEgfs}/jobs/JGDAS_ATMOS_GLDAS. But I can't find this job script in this branch.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA The gldas task (/ecf/scripts/gdas/atmos/init/jgdas_atmos_gldas.ecf) uses ${HOMEgfs}/jobs/JGDAS_ATMOS_GLDAS. But I can't find this job script in this branch.

@HelinWei-NOAA That script lives in the GLDAS repo for GFSv16 still and isn't viewable under a global-workflow clone until we run the link setup step which makes symlinks to the scripts within the component `sorc/*.fd`. Run the `sorc/link_fv3gfs.sh` script after running checkout and build and you'll then see the script under the `jobs/` folder.

Note, for GFSv17 we've absorbed all of the scripts so we won't link to the component checkouts anymore with that version.

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA @aerorahul @yangfanglin So I propose that if cpc gauge data is missing, we just skip the gldas step. 

Change the section in gldas_forcing.sh:

if [ ! -s $cpc ]; then
 echo "FATAL ERROR: GLDAS MISSING $cpc, WILL NOT RUN."
 exit 2
fi

to 
if [ ! -s $cpc ]; then
 echo "GLDAS only runs when CPC gauge data is available "
 exit 0
fi

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA I'm good with this but suggest adding `WARNING` to the message and being more verbose about why the job isn't running. Like this:

```
if [ ! -s $cpc ]; then
  echo "WARNING: GLDAS MISSING ${cpc} and can only run when CPC gauge data is available. Will not run. Exiting."
  exit 0
fi
```

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA Checking in...what are your thoughts on my last suggestion? Do you have additional thoughts or feedback or changes? Thanks!

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA Your suggestion is good. I will make that change.

**Comment by KateFriedman-NOAA** [comment]
> Your suggestion is good. I will make that change.

Great, please let me know when you've made the change and recut the GLDAS tag (you can use the same name). Thanks!

**Comment by HelinWei-NOAA** [comment]
Done. Same tag #
> > Your suggestion is good. I will make that change.
> 
> Great, please let me know when you've made the change and recut the GLDAS tag (you can use the same name). Thanks!

**Comment by KateFriedman-NOAA** [comment]
> Done. Same tag #

Thanks @HelinWei-NOAA ! Please review this PR and, if happy with it, please approve it and I will merge these updates into the auth repo release branch.

**Review by HelinWei-NOAA — APPROVED** [review]

File: ecf/defs/gfs_v16_3.def
Comment (KateFriedman-NOAA) [reply ecf/defs/gfs_v16_3.def:1]: The job will not exit with 2, it will exit with 0 if the data is missing...so the trigger on line 2374 will always be complete regardless of the CPC gauge data being missing or not.

**Pull request merged**
